modules/core/src/convert.dispatch.cpp

```diff
@@ -277,9 +277,21 @@ void Mat::convertTo(OutputArray dst, int type_, double alpha, double beta) const
     int cn = channels();
     int dtype = CV_MAKETYPE(ddepth, cn);
 
-    Mat src = *this;
-    dst.create(dims, size, dtype);
-    Mat dstMat = dst.getMat();
+
+    Mat src;
+    Mat dstMat;
+    if (u != NULL && u->refcount == 1 && data == dst.getMat().data
+        && CV_ELEM_SIZE(stype) == CV_ELEM_SIZE(dtype) && isContinuous())
```

Comment:
this is wrong. When 8u is converted to 8s, for example, `255` should map to `127`. In the case of opposite conversion (8s to 8u) all negative numbers should be converted to 0's. You also don't check `noScale` flag. I suggest to just retain reinterpret() and don't try to patch convertTo(). If reinterpret() works for you, just make the proper test that demonstrates the use of reinterpret().

Reply:
@vpisarev  The function aims to reuse memory while performing data type conversion within the conversion function. such as   ` BinaryFunc func = noScale ? getConvertFunc(sdepth, ddepth) : getConvertScaleFunc(sdepth, ddepth);` 

However, I believe your suggestion is correct: if memory reuse is needed, it's more reasonable to first call the reinterpret function. This is a more correct approach compared to modifying the convertTo function.

Reply:
@vpisarev My understanding and use of OpenCV, as well as my coding skills, indeed have some shortcomings. I am very grateful for your guidance, which has been very beneficial to me.